src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2691,12 +2766,14 @@ static void start_pick_locked(void* arg, grpc_error* error) {
             grpc_error_string(error));
   }
   switch (pick_result) {
-    case LoadBalancingPolicy::PICK_TRANSIENT_FAILURE:
+    case LoadBalancingPolicy::PICK_TRANSIENT_FAILURE: {
       // If we're shutting down, fail all RPCs.
-      if (chand->disconnect_error != GRPC_ERROR_NONE) {
+      grpc_error* disconnect_error =
```

Comment:
Yes, this has to be ACQUIRE because we are unrefing the error and we need to make sure we don't get reordered with compare/exchange. I would add a comment here and for CompareExchange call.